arcflow/utils/bulk_import.py

```diff
@@ -64,6 +67,22 @@ def get_resource_id_from_ead(ead_id, asnake_client):
         print(f'Error searching for resource ID: {e}')
         return None
 
+def check_for_children(repo_id, rid, asnake_client):
+    """Function to check how many top-level children a resource already has. Returns integer value for the number of children or -1 if encounters an error."""
+    try:
+        info = asnake_client.get(f"/repositories/{repo_id}/resources/{rid}/tree/root").json()
+        if 'child_count' not in info:
+            return -1
+
+        return int(info['child_count'])
+    except Exception as e:
+        print(f'Error retrieving child count for resource ID: {e}')
+        return -1
+
+def report_csv_error(report_dict, error_string):
+    """Function to print and log error messages (assumes only one error message)."""
+    report_dict["error"] = error_string
+    print(error_string)
 
 def csv_bulk_import(csv_directory=None, load_type='ao', only_validate='false'):
```

Comment:
Since `retrieve_job_output` modifies the output of `csv_bulk_import`, I recommend to put `retrieve_job_output` inside `csv_bulk_import` to maintain logical cohesion, and add an `asnake_client` parameter to `retrieve_job_output` (this also eliminates the need to instantiate an additional `ASnakeClient`).

Part 2 of 4
```suggestion
def csv_bulk_import(csv_directory=None, load_type='ao', only_validate='false', save_output_files=False):
```